### DISCUSSION TITLE
[Bug]: Popup does not appear

### DISCUSSION THREAD

**Discussion by Gul-liver** [orestbida/cookieconsent#477] [Q&A] [answered]
### Expected Behavior

Hi,
I'm just testing Cookieconsent on a wordpress website, and I've got an issue for displaying the popup.

I try stable release 2.8.9

I put css declaration in header.php:
<link rel="stylesheet" href="/wp-content/themes/hermine-child/cookie-consent/cookieconsent.css" media="print" onload="this.media='all'">

and I call the 2 js files in footer.php:
<script defer src="/wp-content/themes/hermine-child/cookie-consent/cookieconsent.js"></script>
<script defer src="/wp-content/themes/hermine-child/cookie-consent/cookieconsent-init.js"></script>
</body>
</html>

In Firefox console, I've no errors and js files are executed almost as the last js.
But, no popup appear and no cc-cookie is created.

Here my cookieconsent-init.js:

`// obtain plugin
var cc = initCookieConsent();

// run plugin with your configuration
cc.run({
    current_lang: document.documentElement.getAttribute('lang'),
    autoclear_cookies: true,                   // default: false
    page_scripts: true,                        // default: false

    mode: 'opt-in',                          // default: 'opt-in'; value: 'opt-in' or 'opt-out'
    // delay: 0,                               // default: 0
    auto_language: document,                     // default: null; could also be 'browser' or 'document'
    // autorun: true,                          // default: true
    force_consent: false,                   // default: false
	autoclear_cookies: true,
    hide_from_bots: true,                  // default: false
    // remove_cookie_tables: false             // default: false
    // cookie_name: 'cc_cookie',               // default: 'cc_cookie'
    // cookie_expiration: 182,                 // default: 182 (days)
    // cookie_necessary_only_expiration: 182   // default: disabled
    // cookie_domain: location.hostname,       // default: current domain
    // cookie_path: '/',                       // default: root
    // cookie_same_site: 'Lax',                // default: 'Lax'
    // use_rfc_cookie: false,                  // default: false
    // revision: 0,                            // default: 0

    onFirstAction: function(user_preferences, cookie){
        // callback triggered only once
    },

    onAccept: function (cookie) {
        // ...
    },

    onChange: function (cookie, changed_preferences) {
        // ...
    },

	gui_options: {
        consent_modal: {
            layout: 'cloud',               // box/cloud/bar
            position: 'bottom center',     // bottom/middle/top + left/right/center
            transition: 'zoom',           // zoom/slide
            swap_buttons: false            // enable to invert buttons
        },
        settings_modal: {
            layout: 'box',                 // box/bar
            position: 'left',           // left/right
            transition: 'zoom'            // zoom/slide
        }
    },

    languages: {
        'fr': {
            consent_modal: {
                title: 'Nous utilisons des cookies !',
                description: 'Ce site web utilise des cookies essentiels pour assurer son bon fonctionnement et des cookies de suivi d’audience. Ces cookies sont conformes à la législation RGPD. Plus dinformations dans notre <a href="https://lesbijouxdelhermine.fr/politique-de-confidentialite/" class="cc-link">Politique de Confidentialité</a>.',
                primary_btn: {
                    text: 'Accepter',
                    role: 'accept_all'      // 'accept_selected' or 'accept_all'
                },
                secondary_btn: {
                    text: 'Informations',
                    role: 'settings'        // 'settings' or 'accept_necessary'
                }
            },
            settings_modal: {
                title: 'Informations sur les cookies',
                save_settings_btn: 'Enregistrer',
                accept_all_btn: 'Accepter',
				close_btn_label: 'Fermer',
                blocks: [
                    {
                        title: 'Cookies Fonctionnels',
                        description: 'Ces cookies sont nécessaires au bon fonctionnement du site: ils servent, durant votre session, à mémoriser la langue choisie, à mémoriser votre connexion en tant que client, à vous permettre de filtrer l’affichage des produits, à mémoriser votre panier, à enregistrer votre commande. Ces cookies fonctionnels ne sont pas désactivables car cela rendrait impossible votre navigation sur le site. Ces cookies indispensables sont des cookies internes, provenant de notre site lesbijouxdelhermine.fr'
                    }, {
                        title: 'Cookies d’Audience',
                        description: 'L’analyse de votre comportement de navigation est ANONYME, c’est-à-dire que nous ne pourrons pas vous identifier à partir de ces données. Nous recueillons uniquement les données techniques que votre navigateur transfère à notre serveur. Nous collectons à cet effet les données techniquement nécessaires à la fourniture de nos services en ligne et afin d’assurer leur stabilité et leur sécurité conformément à l’article 6, paragraphe 1, phrase 1, du RGPD. Nous ne transférons aucune de ces données à des tiers. Le serveur réalisant cette collecte de données est sous notre seule responsabilité et nous en assurons la gestion. L’outil d’analyse de visites que nous utilisons fait l’objet, par la CNIL, d’une exemption de recueil de consentement des utilisateurs. Cependant, vous pouvez vous opposer à cette analyse. Nous vous informons ici de la façon d’exercer ce droit: <a href="https://lesbijouxdelhermine.fr/politique-de-confidentialite/" class="cc-link">Politique de Confidentialité</a>.',
                        toggle: {
                            value: 'cat-audience-fr',
                            enabled: true,
                            readonly: true
                        }
                    }, 
                ]
            }
        },
        'en': {
            consent_modal: {
                title: 'We use some cookies',
                description: 'This website uses essential cookies to ensure its proper functioning and audience tracking cookies. These cookies comply with the RGPD legislation. More informations are available in our <a href="https://lesbijouxdelhermine.fr/en/privacy-policy/" class="cc-link">Privacy Policy</a>.',
                primary_btn: {
                    text: 'Accept',
                    role: 'accept_all'      //'accept_selected' or 'accept_all'
                },
                secondary_btn: {
                    text: 'Informations',
                    role: 'settings'        //'settings' or 'accept_necessary'
                }
            },
            settings_modal: {
                title: 'Cookie preferences ...',
                save_settings_btn: 'Save',
                accept_all_btn: "Accept all",
                blocks: [
                    {
                        title: 'Functional Cookies.',
                        description: 'These cookies are necessary for the proper functioning of the site: they are used, during your session, to remember the language you have chosen, to remember your connection as a customer, to allow you to filter the display of products, to remember your basket, to record your order. These functional cookies cannot be deactivated as this would make it impossible for you to browse the site. These essential cookies are internal cookies from our website lesbijouxdelhermine.fr'
                    }, {
                        title: 'Audience Cookies',
                        description: 'The analysis of your browsing behaviour is ANONYMOUS, i.e. we cannot identify you from this data. We only collect the technical data that your browser transfers to our server. For this purpose, we collect the data that is technically necessary for the provision of our online services and to ensure their stability and security in accordance with Article 6(1) sentence 1 of the GDPR. We do not transfer any of this data to third parties. We are solely responsible for the server that collects this data and we manage it. The visit analysis tool that we use is subject to an exemption from the collection of user consent by the CNIL. However, you may object to this analysis. We inform you here how to exercise your options in this regard: <a href="https://lesbijouxdelhermine.fr/en/privacy-policy/" class="cc-link">Privacy Policy</a>.',
                        toggle: {
                            value: 'cat-audience-en',
                            enabled: true,
                            readonly: true
                        }
                    }, 
                ]
            }
        }
    }
});
`

The website is https://lesbijouxdelhermine.fr

Could someone help me for understanding what I'm doing wrong ?
Thanks ever so much.
Best regards

PS: I'm aware that's not really a  bug report, but more a call for help ...

### Current Behavior

Modal does not appear

### Steps to reproduce

visiting any page, even after purging cookies

### Version

2.8.9

### On which browser do you see the issue?

Firefox, Other

**Comment by orestbida** [comment, marked as answer]
Hi @Gul-liver,

`cookieconsent.js` and `cookieconsent.css` do not appear on your website. 

They do exist in the hosting server:
https://lesbijouxdelhermine.fr/wp-content/themes/hermine-child/cookie-consent/cookieconsent.js
https://lesbijouxdelhermine.fr/wp-content/themes/hermine-child/cookie-consent/cookieconsent.css

but are not added to the website*.

*Edit: `cookieconsent.js` is loaded, the stylesheet however is missing.

Either it's a cache issue or something wrong with php/wordpress.

**Reply by Gul-liver** [reply]
Thanks for your reply.
Thanks to you, it's now solved. You were right: I had forget that wordpress requires to enqueue css and js properly for working (my previous test was on a static website, very simple), so css was missing and in addition, js order loading was not good.

And in addition, cache issue, as you said ....

Now, everything works. Thanks for your work ! I was looking for a cookie consent application which is simple, reliable, lightweight, I found it !
Best regards,
Bertrand

For those who could be interested, adding this in functions.php works well:

`/*********************************************************Scripts Cookie Consent *****************/
function import_scripts_and_styles_cookieconsent() {
wp_enqueue_style( 'cookieconsent_css', get_stylesheet_directory_uri() . '/cookie-consent/cookieconsent.css', false, '1.0', 'all');
wp_enqueue_script( 'cookieconsent_js', get_stylesheet_directory_uri() . '/cookie-consent/cookieconsent.js', array(), 1.0, true );
}
add_action('wp_enqueue_scripts', 'import_scripts_and_styles_cookieconsent', 20);

function import_scripts_and_styles_cookieconsent_2() {
wp_enqueue_script( 'cookieconsent_init_js', get_stylesheet_directory_uri() . '/cookie-consent/cookieconsent-init.js', array(), 1.0, true );
}
add_action('wp_enqueue_scripts', 'import_scripts_and_styles_cookieconsent_2', 25);
/*********************************************************Scripts Cookie Consent *****************/`

**Reply by M4rtyCZE** [reply]
Works like a charm, thanks guys!
